### DISCUSSION TITLE
[CONTRIBUTION] Speech Dataset Generator

### DISCUSSION THREAD

**Discussion by davidmartinrius** [espnet/espnet#5689] [Show and tell]
Hi everyone!

My name is David Martin Rius and I have just published this project on GitHub: https://github.com/davidmartinrius/speech-dataset-generator/

Now you can **create datasets automatically** with any audio or lists of audios.

I hope you find it useful.

### Here are the key functionalities of the project:

1. **Dataset Generation:** Creation of multilingual datasets with Mean Opinion Score (MOS).

2. **Silence Removal:** It includes a feature to remove silences from audio files, enhancing the overall quality.

3. **Sound Quality Improvement:** The project focuses on improving the quality of the audio.

4. **Audio Segmentation:** It can segment audio files within specified second ranges.

5. **Transcription:** The project transcribes the segmented audio, providing a textual representation.

6. **Gender Identification:** It identifies the gender of each speaker in the audio.

7. **Pyannote Embeddings:** Utilizes pyannote embeddings for speaker detection across multiple audio files.

8. **Automatic Speaker Naming:** Automatically assigns names to speakers detected in multiple audios.

9. **Multiple Speaker Detection:** Capable of detecting multiple speakers within each audio file.

10. **Store speaker embeddings:** The speakers are detected and stored in a Chroma database, so you do not need to assign a speaker name.

11. **Syllabic and words-per-minute metrics**

Feel free to explore the project at https://github.com/davidmartinrius/speech-dataset-generator

**Comment by sw005320** [comment]
Thanks for letting us know, and congratulations on your great achievement!

**Reply by davidmartinrius** [reply]
Thank you @sw005320 !

**Comment by davidmartinrius** [comment]
@sw005320 I added a new TODO for espnet audio enhancer in [speech-dataset-generator](https://github.com/davidmartinrius/speech-dataset-generator/blob/main/README.md#audio-enhancers)

I consider it can be very useful. It works very well when separating multiple speakers from an audio.
